Expand mqtt command send notification to include more variables.

brjhaverkamp opened the pull request:
This allows for sending notifications to a single subsystem or device
(My usecase is sending a message over FCM to only my mobile)

Commit:
Expand mqtt command send notification to include more variables.
This allows for sending notifications to a single subsystem or device
(My usecase is sending a message over FCM to only my mobile)

Commit:
Expand mqtt command sendnotification to include more variables.
This allows for sending notifications to a single subsystem or device
(My usecase is sending a message over FCM to only my mobile)

gizmocuz commented on hardware/MQTT.cpp:
```diff
@@ -482,8 +482,22 @@ void MQTT::on_message(const struct mosquitto_message* message)
 		{
-			std::string subject, body, sound;
+			uint64_t idx=0;
+			std::string name=std::string(""), subsystems=NOTIFYALL, subject, body, extradata=std::string(""), sound;
```
Could you place subsystem on a separate line as you initialize this
You do not have to use std::string(""), a string is initialized empty by default.

Commit:
Merge branch 'mqttsendnotification' of https://github.com/brjhaverkamp/domoticz into mqttsendnotification

Commit:
Aditional changes as requested in discussion on pull request
Merge branch 'mqttsendnotification' of https://github.com/brjhaverkamp/domoticz into mqttsendnotification

Commit:
Merge branch 'mqttsendnotification' of https://github.com/brjhaverkamp/domoticz into mqttsendnotification

Commit:
Cleaned up the previous change as requested

brjhaverkamp commented:
I did the update, but made bit of a mess on commiting and reverting  
(first time I worked with a github pull request, learning on the fly)
Let me know if this is now OK, or what I should do to clean it up.

Bert

gizmocuz commented:
Looking great!! Thanks for this !!!

rwaaren commented:
@brjhaverkamp

Below is what is in [the wiki MQTT to domoticz ](https://www.domoticz.com/wiki/MQTT#MQTT_to_Domoticz ) now. 

{"command": "sendnotification", "subject": "My Subject", "body": "My Body", "priority": 0, "sound": "mysound" }

Can you reply with what is should be now? I will update the Wiki accordingly.

Or even better update the wiki yourself (You can send me a PM on the forum to request a userid to do that)

brjhaverkamp commented:
Hi and thanks for reaching out. It was on my ToDo-list to find how I could update the wiki on this.
I'll PM you for the id.

rwaaren commented:
@brjhaverkamp You have a reply with a request on the forum.
